src/Components/Web.JS/src/Platform/WebAssemblyProgressService.ts

```diff
@@ -0,0 +1,25 @@
+type SetProgressFunction = (resourcesTotal: number, resourcesLoaded: number) => void;
+
+export class WebAssemblyProgressService {
+    private resourcesTotal: number = 0;
+    private resourcesLoaded: number = 0;
+    private observerList = new Set<SetProgressFunction>();
+
+    private notifyAllObservers() {
+        this.observerList.forEach(sp => sp(this.resourcesTotal, this.resourcesLoaded));
+    }
+
+    setTotalResources(resourcesTotal: number) {
+        this.resourcesTotal += resourcesTotal;
+        this.notifyAllObservers();
+    }
+
+    resourceLoaded() {
+        this.resourcesLoaded++;
+        this.notifyAllObservers();
+    }
+
+    attach(observer: SetProgressFunction): void {
+        this.observerList.add(observer);
+    }
```

Comment:
Can we add TSDoc documentation to these public methods like we have elsewhere in `Web.JS`? Also, for consistency, can we ensure `resourceLoaded` / `setTotalResources` have explicit return types (`void`)